### PR TITLE
Fix loading spaces with api_name=False

### LINKED ISSUE #4884: `Client` doesn't work properly when a Space has a method with `api_name=False`
### Describe the bug

Originally reported [here](https://huggingface.co/spaces/hysts/Shap-E/discussions/18).

If a Space had a method with `api_name=False`, the following error occurs when calling it with `Client`:
```
TypeError: can only concatenate str (not "bool") to str
```

I think this error occurs when calling a Space with `gr.Examples`.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

When I use `Client` to call [this Space](https://huggingface.co/spaces/hysts-debug/client-bool-api-name-error)
```python
import gradio as gr

with gr.Blocks() as demo:
    text = gr.Text()
    out = gr.Text()
    btn = gr.Button()
    text.submit(fn=lambda x: x, inputs=text, outputs=out, api_name=False)
    btn.click(fn=lambda x: x, inputs=text, outputs=out, api_name='run')
demo.queue().launch()
```
with the following code
```python
from gradio_client import Client

client = Client('hysts-debug/client-bool-api-name-error')
out = client.predict('aaa', api_name='/run')
```
I get the following error.
```
Traceback (most recent call last):
  File "/tmp/temp/run.py", line 4, in <module>
    out = client.predict('aaa', api_name='/run')
  File "/home/ubuntu/.virtualenvs/py31011/lib/python3.10/site-packages/gradio_client/client.py", line 285, in predict
    return self.submit(*args, api_name=api_name, fn_index=fn_index).result()
  File "/home/ubuntu/.virtualenvs/py31011/lib/python3.10/site-packages/gradio_client/client.py", line 313, in submit
    inferred_fn_index = self._infer_fn_index(api_name, fn_index)
  File "/home/ubuntu/.virtualenvs/py31011/lib/python3.10/site-packages/gradio_client/client.py", line 555, in _infer_fn_index
    if "/" + config_api_name == api_name:
TypeError: can only concatenate str (not "bool") to str
```

If I remove the method with `api_name=False`, the error doesn't occur.

In the case of the [Shap-E Space](https://huggingface.co/spaces/hysts/Shap-E), it doesn't have methods with `api_name=False` explicitly, but as it has `gr.Examples`, I guess its method has `api_name=False` implicitly and causing this error.

If I change the code of the Space to the following, the same error occurs.
```python
import gradio as gr

with gr.Blocks() as demo:
    text = gr.Text()
    out = gr.Text()
    btn = gr.Button()
    gr.Examples(examples=['aaa', 'bbb'], inputs=text)
    btn.click(fn=lambda x: x, inputs=text, outputs=out, api_name='run')
demo.queue().launch()
```

### Screenshot

_No response_

### Logs

_No response_

### System Info

```shell
gradio==3.36.1
gradio_client==0.2.8
```

### Severity

Blocking usage of gradio

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Closes #4884 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests & Changelog

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

1. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`
  
1. Unless the pull request is labeled with the "no-changelog-update" label by a maintainer of the repo, all pull requests must update the changelog located in `CHANGELOG.md`:

Please add a brief summary of the change to the Upcoming Release section of the `CHANGELOG.md` file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

### CODE REVIEW

File: client/python/gradio_client/client.py [hunk 1]
```diff
@@ -550,7 +550,7 @@ def _infer_fn_index(self, api_name: str | None, fn_index: int | None) -> int:
         if api_name is not None:
             for i, d in enumerate(self.config["dependencies"]):
                 config_api_name = d.get("api_name")
-                if config_api_name is None:
+                if config_api_name is None or config_api_name is False:
```
Comment (freddyaboulton) [inline client/python/gradio_client/client.py:553]: Not writing as `if not config_api_name` cause the `api_name` can theoretically be `""` which evaluates to false.

**Comment by abidlabs** [comment]
Nice! Code looks good + can confirm that the original issue has been fixed.

**Review by abidlabs — APPROVED** [review]

File: client/python/gradio_client/client.py (hunk 1 above)
Comment (hysts) [reply client/python/gradio_client/client.py:553]: @freddyaboulton 
Hmm, but I think `api_name=''` should be considered invalid.

Looks like use via API page becomes weird when it has a method with `api_name=''`.
In the case of the [Space](https://huggingface.co/spaces/gradio-tests/client-bool-api-name-error) used for testing this PR, the API page is empty.


Also, if a Space has only one method and whose `api_name` is `''`, then the API page will become like this:

Though it says there's one endpoint, but no API was shown.

I think we should change to make empty API name invalid (in another PR?).

Comment (freddyaboulton) [reply client/python/gradio_client/client.py:553]: Yea we can solve that issue in a separate PR!

**Comment by freddyaboulton** [comment]
Thanks for the reviews @hysts and @abidlabs !

**Pull request merged**